controller/appcontroller.go

```diff
@@ -1389,15 +1388,38 @@ func (ctrl *ApplicationController) processAppRefreshQueueItem() (processNext boo
 		localManifests = opState.Operation.Sync.Manifests
 	}
 
-	revision := app.Spec.Source.TargetRevision
-	if comparisonLevel == CompareWithRecent {
-		revision = app.Status.Sync.Revision
-	}
+	revisions := make([]string, 0)
+	sources := make([]appv1.ApplicationSource, 0)
+
+	hasMultipleSources := app.Spec.Sources != nil && len(app.Spec.Sources) > 0
 
+	// If we have multiple sources, we use all the sources under `sources` field and ignore source under `source` field.
```

Comment:
Would a viable option be to normalise `sources` by instead appending a single `source` to `sources` if that is present?

Reply:
I think we'd like to completely abandon the `source` field in 3.0, so probably better to encourage folks to just use `sources`.